### PR TITLE
feat(js): Add docs for generic pool integration

### PR DESCRIPTION
**Pull request by AbhiPrasad**
## DESCRIBE YOUR PR

Documents https://github.com/getsentry/sentry-javascript/pull/13465

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/javascript/common/configuration/integrations/genericpool.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: Mongoose
```
Comment (vivianyentran) [inline docs/platforms/javascript/common/configuration/integrations/genericpool.mdx:2]:
```suggestion
title: Generic Pool
```

File: docs/platforms/javascript/common/configuration/integrations/genericpool.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+_Import name: `Sentry.genericPoolIntegration`_
+
+This integration is enabled by default when performance monitoring is enabled. If you'd like to modify your default integrations, read [this](./../#modifying-default-integrations).
+
+The `genericPoolIntegration` adds instrumentation for the `generic-pool` library to capture spans using [`@opentelemetry/instrumentation-generic-pool`](https://www.npmjs.com/package/@opentelemetry/instrumentation-generic-pool).
+
+## Supported Versions
+
+- `mongoose`: `>=2.0.0 <4`
```
Comment (vivianyentran) [inline docs/platforms/javascript/common/configuration/integrations/genericpool.mdx:38]: Just want to double check that this should be mongoose

**Review by vivianyentran — CHANGES_REQUESTED** [review]

File: docs/platforms/javascript/common/configuration/integrations/genericpool.mdx (hunk 2 above)
Comment (AbhiPrasad) [reply docs/platforms/javascript/common/configuration/integrations/genericpool.mdx:38]: it should not! sorry for that.

File: docs/platforms/javascript/common/configuration/integrations/genericpool.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (AbhiPrasad) [inline docs/platforms/javascript/common/configuration/integrations/genericpool.mdx:38]:
```suggestion
- `generic-pool`: `>=2.0.0 <4`
```

**Pull request merged**